pkg/server/event_handler_test.go

```diff
@@ -325,18 +388,40 @@ func newTestServer(handler func(w http.ResponseWriter, r *http.Request)) (*http.
 	}
 }
 
+// Returns fake metadata that include github resource info.
+func getFakeMetadata() []byte {
+	return []byte(`{
+		"metadata": {
+		  "git-commit": "test-github-commit",
+		  "triggered-timestamp": "2023-04-25T17:44:57",
+		  "git-workflow-sha": "test-workflow-sha",
+		  "git-workflow": "test-workflow",
+		  "git-repo": "test-github-repo"
+		}
+	  }`)
+}
+
 // Returns a fake http func that writes the data in http response.
 func handleObjectRead(t *testing.T, data []byte) func(w http.ResponseWriter, r *http.Request) {
 	t.Helper()
 
 	return func(w http.ResponseWriter, r *http.Request) {
-		if r.URL.String() != "/foo/bar" {
+		switch r.URL.Path {
+		// This is for getting object info
+		case "/foo/bar":
+			_, err := w.Write(data)
+			if err != nil {
+				t.Fatalf("failed to write response for object info: %v", err)
+			}
+		// This is for getting object's metadata
+		case "/storage/v1/b/foo/o/bar":
+			_, err := w.Write(getFakeMetadata())
+			if err != nil {
+				t.Fatalf("failed to write response for object metadata: %v", err)
+			}
```

Comment:
Do we still need this given we already get metadata from pubsub message?

Reply:
Removed.